### PR TITLE
Added moment.range.fromInterval.

### PR DESCRIPTION
**Pull request by rosshadden**

### CODE REVIEW

File: lib/moment-range.js [hunk 1]
```diff
@@ -353,6 +353,27 @@ moment.range = function(start, end) {
...
 /**
+ * Build a date range between a specified interval and now (or a specified date).
+ *
+ * @param {String} interval The type of interval
+ * @param {Number} [count=1] The number of intervals
+ * @param {(Moment|Date} [date=moment()] The date to use
+ *
+ * @return {!DateRange}
+ */
+moment.range.fromInterval = function(interval, count = 1, date = moment()) {
```
Comment (andyo729) [inline lib/moment-range.js:365]: Can you write a test for the new method?

File: lib/moment-range.js [hunk 2]
```diff
@@ -353,6 +353,27 @@ moment.range = function(start, end) {
...
+  if (!moment.isMoment(date)) date = moment(date);
```
Comment (andyo729) [inline lib/moment-range.js:366]: After this, check `date.isValid()`

File: lib/moment-range.js [hunk 3]
```diff
@@ -353,6 +353,27 @@ moment.range = function(start, end) {
...
+
+  let date2 = date.clone().add(count, interval);
+  let dates = [];
+
+  dates.push(moment.min(date, date2));
```
Comment (andyo729) [inline lib/moment-range.js:373]: Isn't `date` always the `min` and `date2` always the `max`?

File: lib/moment-range.js [hunk 4]
```diff
@@ -353,6 +353,27 @@ moment.range = function(start, end) {
...
+  dates.push(moment.max(date, date2));
+
+  return new DateRange(dates);
```
Comment (andyo729) [inline lib/moment-range.js:376]: I think you could just `return new DateRange([date, date2]);` and remove the several lines above that are building up the array.

**Comment by andyo729** [comment]
@rosshadden Few updates and test

File: lib/moment-range.js (hunk 3 above)
Comment (rawsp33d) [inline lib/moment-range.js:373]: No.  The code that adds them in the order of `[ min, max ]` is what makes negative intervals work.

File: lib/moment-range.js (hunk 2 above)
Comment (rawsp33d) [inline lib/moment-range.js:366]: Yeah, I mean I was trying to be consistent with his library (which is hard!), and he never calls that.  But it _is_ a great idea.  So I'll do it and he can learn from it!

File: lib/moment-range.js (hunk 4 above)
Comment (rawsp33d) [inline lib/moment-range.js:376]: NOPE, see above PR comment.

File: lib/moment-range.js (hunk 3 above)
Comment (rawsp33d) [inline lib/moment-range.js:373]: I'll add an inline comment about it, to make that more clear to everyone.

**Pull request merged**